### PR TITLE
check if domain is blocked or not by checking it against list.preEven…

### PR DESCRIPTION
**Pull request by PromoFaux**
_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

**By submitting this pull request, I confirm the following (please check boxes, eg [X]):**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

- [x] 10 (very familiar)

---

Alternative way of deciding if domain is Pi-holed or not. May slow down on Pi.. not yet tested...

### CODE REVIEW

**Comment by PromoFaux** [comment]
17,683 queries
Pi 2 Model B v1.1 `master` branch 11.59 seconds:
image

Pi 2 Model B v1.1 `fixTheQueryLog` branch  13.66 seconds:
image

File: data.php [hunk 1]
```diff
@@ -188,16 +190,10 @@ function getAllQueries($orderBy) {
...
-            }
-            elseif ((substr($tmp, 0, 9) == "forwarded" || $exploded[count($exploded)-3] == "pi.hole" || $exploded[count($exploded)-3] == $hostname) && $showpermitted){
-              $status="OK";
-            }
-            elseif ((substr($tmp, strlen($tmp) - 12, 12)  == "gravity.list"  && $exploded[count($exploded)-5] != "read") && $showblocked){
-              $status="Pi-holed";
+                $type = substr($exploded[count($exploded)-4], 6, -1);
+                $domain = $exploded[count($exploded)-3];
+                $client = $exploded[count($exploded)-1];
+                $status = isset($gravity_domains[$domain]) ? "Pi-holed" : "OK";
```
Comment (DL6ER) [inline data.php]: Have to check against `$showpermitted` and `$showblocked` before adding to the list

File: data.php [hunk 2]
```diff
@@ -188,16 +190,10 @@ function getAllQueries($orderBy) {
...
             }
 
             if ( $status != ""){
```
Comment (DL6ER) [inline data.php]: The `if` is obsolete. The code above will always set a `$status` and the two `if` clauses can be combined

**Comment by DL6ER** [comment]
Go ahead if you think my changes are fine. I tested it with > 10.000 queries and it works fine for me.

Approved

**Comment by PromoFaux** [comment]
OK yeah, this is actually about a second faster than mine ;)

**Pull request merged**